node_group/aws_managed/main.tf

```diff
@@ -33,13 +30,26 @@ resource "aws_launch_template" "this" {
   image_id      = var.ami_id
   # # Set on node group instead
   # instance_type = var.launch_template_instance_type
-  key_name               = var.key_name
-  user_data              = var.before_cluster_joining_userdata
-  vpc_security_group_ids = var.vpc_security_group_ids
-
+  key_name                = var.key_name
+  user_data               = var.before_cluster_joining_userdata
+  vpc_security_group_ids  = var.vpc_security_group_ids
   disable_api_termination = var.disable_api_termination
   kernel_id               = var.kernel_id
   ram_disk_id             = var.ram_disk_id
+  default_version         = var.update_launch_template_default_version ? var.launch_template_default_version : null
+
+  tag_specifications {
+    resource_type = "instance"
+    tags          = var.launch_template_tags
+  }
+
+  instance_market_options {
+    market_type = var.instance_market_options.market_type
+
+    spot_options {
+      max_price = var.instance_market_options.spot_options.max_price
+    }
+  }
```

Comment:
Please use Dynamic Block for these parameters with no hardcoded values.

Reply:
this is fixed